ProcessList.c

```diff
@@ -116,22 +116,29 @@ int ProcessList_size(ProcessList* this) {
    return (Vector_size(this->processes));
 }
 
-// ProcessList_updateTreeSetLayer sort this->treeSet,
+// ProcessList_updateTreeSetLayer sorts this->displayTreeSet,
 // relying only on itself.
 //
-// 'newTree' is a temporaraly Hashtable which at the end of the function
-// will hold a renovated 'treeSet'.
-// It's necessary to use a temporaraly Hashtable as it's not save to traverse this->treeSet
-// and at the same time make changes in it.
+// Algorithm
+//
+// The algorithm is based on `Depth-first search`,
+// even though `Breadth-first search` approach may be more eficient from first perspetive,
+// after comparision it may be not. As it's not save to not update the tree structure after getting deeper.
+// If it would be save that approach would likely bring an advantage in performance.
 //
-// It could though be removed with sort by TREE_SET if we could savely update this->processes in place.
+// Each call of the function looks for a 'layer'. A 'layer' is a list of processes with the same deep.
+// It sorts a list. Then it runs the function recursevely for each element of sorted list.
+// And then it updates settings of processes.
```

Comment:
```suggestion
// First it sorts a list. Then it runs the function recursively for each element of the sorted list.
// After that it updates the settings of processes.
```
With settings you probably mean "position"/"order" kind of information of each process?

Reply:
yes `lhs`, `rhs`, `deep` fields